get_team_roles/team_roles.py

```diff
@@ -1,39 +1,64 @@
 import argparse
 import sys
 import pdpyras
+import csv
 
-def get_teams(session, comma_separated):
-    if comma_separated:
-        sys.stdout.write("Team ID, Team Name, User ID, User name, Team role\n")
+def get_teams(session):
+    if args.comma_separated:
+        sys.stdout.write("Team ID', 'Team Name', 'User ID', 'User name', 'Email', 'Base role', 'Team role'\n")
+    elif args.csv_file:
+        tf = csv.writer(args.csv_file)
+        sys.stdout.write("Creating / Clearing : {}\n".format(args.csv_file.name))
+        tf.writerow(['Team ID', 'Team Name', 'User ID', 'User name', 'Email', "Base role", 'Team role'])
     try:
         for team in session.iter_all('teams'):
-            get_team_members(team['id'], team['name'], session, comma_separated)
+            get_team_members(team['id'], team['name'], session)
     except pdpyras.PDClientError as e:
         raise e
 
-def get_team_members(team_id, team_name, session, comma_separated):
+def get_team_members(team_id, team_name, session):
     try:
-        for member in session.iter_all('teams/{}/members'.format(team_id)):
-            if comma_separated:
-                sys.stdout.write("{}, {}, {}, {}, {}\n".format(team_id, team_name, member['user']['id'], member['user']['summary'], member['role']))
+        for member in session.iter_all('teams/{}/members'.format(team_id), paginate=True):
+            user = get_user_data(member['user']['id'], session)
+            if args.comma_separated:
+                sys.stdout.write("{}, {}, {}, {}, {}\n".format(team_id, team_name, user['id'], user['name'], user['email'], user['role'], member['role']))
+            elif args.csv_file:
+                tf = csv.writer(args.csv_file)
+                tf.writerow([team_id, team_name, user['id'], user['name'], user['email'], user['role'], member['role']])
+                sys.stdout.write("Added to {}: {}, {}, {}, {}, {}\n".format(args.csv_file.name, team_id, team_name, user['id'], user['name'], user['email'], user['role'], member['role']))
+
             else:
                 sys.stdout.write("Team ID: {}\n".format(team_id))
                 sys.stdout.write("Team Name: {}\n".format(team_name))
-                sys.stdout.write("User ID: {}\n".format(member['user']['id']))
-                sys.stdout.write("User name: {}\n".format(member['user']['summary']))
+                sys.stdout.write("User ID: {}\n".format(user['id']))
+                sys.stdout.write("User name: {}\n".format(user['name']))
+                sys.stdout.write("User email: {}\n".format(user['email']))
+                sys.stdout.write("Base role: {}\n".format(user['role']))
                 sys.stdout.write("Team role: {}\n".format(member['role']))
                 sys.stdout.write("-----\n")
     except pdpyras.PDClientError as e:
         print("Could not get team members for team {} {}".format(team_name, team_id))
         raise e
 
-
+def get_user_data(user_id, session):
+    if user_id in Users:
```

Comment:
having a variable implicitly passed to a function generally isn't best practice. Same goes for `args`. Might be worth looking at how the code could be restructured to make what it's doing more explicit.